samples/matter/thermostat/src/temperature_manager.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+
+#include "temperature_manager.h"
+#include "app_config.h"
+#include "app_task.h"
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace chip;
+using namespace ::chip::DeviceLayer;
+
+constexpr EndpointId kThermostatEndpoint = 1;
+
+namespace ThermostatAttributes = chip::app::Clusters::Thermostat::Attributes;
+
+uint8_t ReturnCompleteValue(int16_t Value)
+{
+    uint8_t result = Value/100; // Convert and return only complete part of value to printable type 
+    return result;
+
+}
+
+uint8_t ReturnReminderValue(int16_t Value)
+{
+
+    uint8_t remainder = (Value%100 + 5)/10; // Converts and returns only reminder part of value to printable type. 
+                                            // This formula rounds reminder value to one significant figure
+    return remainder;
+
+}
+
+
+CHIP_ERROR TemperatureManager::Init()
+{
+    app::DataModel::Nullable<int16_t> temp;
+
+
+
+    PlatformMgr().LockChipStack();
+    ThermostatAttributes::LocalTemperature::Get(kThermostatEndpoint, temp);
+    ThermostatAttributes::OccupiedCoolingSetpoint::Get(kThermostatEndpoint, &mHeatingCelsiusSetPoint);
+    ThermostatAttributes::OccupiedHeatingSetpoint::Get(kThermostatEndpoint, &mCoolingCelsiusSetPoint);
+    ThermostatAttributes::SystemMode::Get(kThermostatEndpoint, &mThermMode);
+    PlatformMgr().UnlockChipStack();;
+
+
+    mCurrentTempCelsius     = temp.Value();
+
+
+
+    LogThermostatStatus();
+
+    return CHIP_NO_ERROR;
+}
+
+void TemperatureManager::LogThermostatStatus()
+{
+
+    LOG_INF("Thermostat Status - M:%d T:%d,%d'C H:%d,%d'C C:%d,%d'C", GetMode(), 
+                                                            ReturnCompleteValue(GetCurrentTemp()),  ReturnReminderValue(GetCurrentTemp()),
```

Comment:
To stick c++ style you can declare ReturnReminderValue & ReturnCompleteValue only within this file by declaring them as `static` or putting them in an anonymous namespace, or just create lambdas here since they are used only in this method.